default/python/AI/ColonisationAI.py

```diff
@@ -983,46 +999,46 @@ def evaluate_planet(planet_id, mission_type, spec_name, detail=None):
         mining_bonus = 0
         per_ggg = 10
 
-        got_asteroids = False
-        got_owned_asteroids = False
-        local_gg = False
-        got_owned_gg = False
+        asteroid_factor = 0.0
+        gg_factor = 0.0
         ast_shipyard_name = ""
         if system and FocusType.FOCUS_INDUSTRY in species.foci:
             for pid in [temp_id for temp_id in system.planetIDs if temp_id != planet_id]:
                 p2 = universe.getPlanet(pid)
                 if p2:
                     if p2.size == fo.planetSize.asteroids:
-                        got_asteroids = True
+                        if p2.owner==empire.empireID:
+                            asteroid_factor = 1.0
+                        elif p2.unowned:
+                            asteroid_factor = 0.5
+                        elif pid in prospective_invasion_targets:
+                            asteroid_factor = character.secondary_valuation_factor_for_invasion_targets()
+                        else:
+                            asteroid_factor = 0.0
                         ast_shipyard_name = p2.name
-                        if p2.owner != -1:
-                            got_owned_asteroids = True
                     if p2.size == fo.planetSize.gasGiant:
-                        local_gg = True
-                        if p2.owner != -1:
-                            got_owned_gg = True
-        if got_asteroids:
+                        if p2.owner==empire.empireID:
+                            gg_factor = 1.0
+                        elif p2.unowned:
+                            gg_factor = 0.5
+                        elif pid in prospective_invasion_targets:
+                            gg_factor = character.secondary_valuation_factor_for_invasion_targets()
+                        else:
+                            gg_factor = 0.0
+        if asteroid_factor > 0.0:
             if tech_is_complete("PRO_MICROGRAV_MAN") or "PRO_MICROGRAV_MAN" in empire_research_list[:10]:
-                if got_owned_asteroids:  # can be quickly captured
-                    flat_industry += 5  # will go into detailed industry projection
-                    detail.append("Asteroid mining ~ %.1f" % (5 * discount_multiplier))
-                else:  # uncertain when can be outposted
-                    asteroid_bonus = 2.5 * discount_multiplier  # give partial value
-                    detail.append("Asteroid mining %.1f" % (5 * discount_multiplier))
+                flat_industry += 5 * asteroid_factor  # will go into detailed industry projection
```

Comment:
I figured the existing implementation updated the ```asteroid_bonus``` rather than the ```flat_industry``` to explicitly account for the fact that a non-owned asteroid would not add the industry bonus directly so it shouldn't be treated as a real bonus here (and in the projection).

Reply:
The previous code had given a half-value when the asteroids were not owned by the current empire, because it's "uncertain when can be outposted [or invaded]".  It seems to me that the treatment is mostly just generalized some now-- the main differences are that different aggression AIs can now weight that uncertainty/delay differently, and the calculation is I think improved by changing the buckets.  

As far was which bucket it goes into, it seemed best to me to make the 'asteroid_bonus' be focused on the non-industrial (i.e., currently shipbuilding) benefits from asteroids, and let the risk-adjusted industry bonus be have it's expected value be calculated along with other industry bonuses, which takes into account the time needed for meters to grow, rather than just leave it at the simplistic base*discount_multiplier treatment (which was essentially just treating it as if the meter instantly gets the full (or in this case, half) benefit of the bonus.  The difference becomes more significant the more other bonuses to industry there are.

The only way I really see to improve that treatment, would be rather than just risk-scaling the base bonus, we did one calc without the contingent bonuses (i.e. stemming from planets not owned by the present empire) at all, and another calc with all the contingent bonuses being given full value, and then we do the risk scaling on the difference between those two values.  I suppose we would also need to keep track of different sources of contingency maybe having different risk scales associated with them, and then wind up combining multiple different contingent projections.  That sounds to me like a good treatment, but a bit more then I can take on right now.

Reply:
Thinking some more about it, I have no more objections to "mess up" the projected industry value. The AI will generally pretty greedily outpost asteroids in that system anyway. 

The industry projection should probably even be extended at some point in the (distant) future to also account for (speculatively) to-be-finished techs in the time period of projection.